source/extensions/network/dns_resolver/cares/dns_impl.h

```diff
@@ -68,18 +68,27 @@ class DnsResolverImpl : public DnsResolver, protected Logger::Loggable<Logger::I
   friend class DnsResolverImplPeer;
   class PendingResolution : public ActiveDnsQuery {
   public:
+    // Network::ActiveDnsQuery
     void cancel(CancelReason reason) override {
       // c-ares only supports channel-wide cancellation, so we just allow the
       // network events to continue but don't invoke the callback on completion.
       // TODO(mattklein123): Potentially use timeout to destroy and recreate the channel.
       cancelled_ = true;
       cancel_reason_ = reason;
     }
+
+    // Network::ActiveDnsQuery
+    void addTrace(uint8_t) override {}
+
+    // Network::ActiveDnsQuery
+    const std::vector<Trace>& getTraces() override { return traces_; }
```

Comment:
makes me wonder if we should make getTraces an optref to both avoid the vector for non-implemented cases and make it clear where there is support, WDYT?